src/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+# libtool versioning
+# See http://sources.redhat.com/autobook/autobook/autobook_91.html#SEC91 for details
+
+## increment if the interface has additions, changes, removals.
+set(LT_CURRENT 21)
+
+## increment any time the source changes; set 0 to if you increment CURRENT
+set(LT_REVISION 0)
+
+## increment if any interfaces have been added; set to 0
+## if any interfaces have been changed or removed. removal has
+## precedence over adding, so set to 0 if both happened.
+set(LT_AGE 13)
+
+math(EXPR LT_CURRENT_MINUS_AGE "${LT_CURRENT} - ${LT_AGE}")
+
+set(SONAME ${LT_CURRENT_MINUS_AGE}.${LT_AGE}.${LT_REVISION})
+set(SOVERSION ${LT_CURRENT_MINUS_AGE})
+
+if (ENABLE_CCE)
+    add_subdirectory("CCE")
+endif()
+add_subdirectory("common")
+add_subdirectory("CPE")
+add_subdirectory("CVE")
+add_subdirectory("CVSS")
+add_subdirectory("DS")
+add_subdirectory("OVAL")
+add_subdirectory("SCE")
+add_subdirectory("source")
+add_subdirectory("XCCDF")
+add_subdirectory("XCCDF_POLICY")
+
+set(OBJECTS_TO_LINK_AGAINST
+    $<TARGET_OBJECTS:common_object> $<TARGET_OBJECTS:cpe_object> $<TARGET_OBJECTS:cve_object> $<TARGET_OBJECTS:cvss_object>
+    $<TARGET_OBJECTS:ds_object> $<TARGET_OBJECTS:oval_object> $<TARGET_OBJECTS:ovaladt_object> $<TARGET_OBJECTS:ovalcmp_object>
+    $<TARGET_OBJECTS:ovalresults_object> $<TARGET_OBJECTS:oscapsource_object> $<TARGET_OBJECTS:rbt_object>
+    $<TARGET_OBJECTS:seap_object> $<TARGET_OBJECTS:xccdf_object> $<TARGET_OBJECTS:xccdfPolicy_object>
+)
```

Comment:
why do you link to object files instead of the libraries themselves? both approaches are valid.

IMO either build the libraries but then also use them,
or use object files and avoid building libraries altogether. Right now the libraries are built and wasted.

Reply:
I have no idea why it was done in this way. I think the libraries don't have to be build.

Reply:
Morevoer it is not always, eg. in CVSS directory it builds only the objects.

Reply:
@mpreisler I have looked into this and I have found that the libraries are used when the probes are build. I wanted to remove it and build only objects, but then I wasn't able to get any probe built. :-(

Reply:
So the probes static link with the libraries? Why don't they link to libopenscap.so dynamically?